neural_tangents/utils/test_utils.py

```diff
@@ -102,8 +105,42 @@ def assertAllClose(
       canonicalize_dtypes=True):
     if isinstance(x, Kernel):
       self.assertIsInstance(y, Kernel)
-      x_dict = dataclasses.asdict(x)
-      y_dict = dataclasses.asdict(y)
+      x_dict = {
+        "nngp": x.nngp,
+        "ntk": x.ntk,
+        "cov1": x.cov1,
+        "cov2": x.cov2,
+        "x1_is_x2": x.x1_is_x2,
+        "is_gaussian": x.is_gaussian,
+        "is_reversed": x.is_reversed,
+        "is_input": x.is_input,
+        "diagonal_batch": x.diagonal_batch,
+        "diagonal_spatial": x.diagonal_spatial,
+        "shape1": x.shape1,
+        "shape2": x.shape2,
+        "batch_axis": x.batch_axis,
+        "channel_axis": x.channel_axis,
+        "mask1": x.mask1,
+        "mask2": x.mask2
+      }
+      y_dict = {
+        "nngp": y.nngp,
+        "ntk": y.ntk,
+        "cov1": y.cov1,
+        "cov2": y.cov2,
+        "x1_is_x2": y.x1_is_x2,
+        "is_gaussian": y.is_gaussian,
+        "is_reversed": y.is_reversed,
+        "is_input": y.is_input,
+        "diagonal_batch": y.diagonal_batch,
+        "diagonal_spatial": y.diagonal_spatial,
+        "shape1": y.shape1,
+        "shape2": y.shape2,
+        "batch_axis": y.batch_axis,
+        "channel_axis": y.channel_axis,
+        "mask1": y.mask1,
+        "mask2": y.mask2
+      }
```

Comment:
Would it be possible to make the `Kernel` class a `@dataclasses.dataclass` (i.e. https://docs.python.org/3/library/dataclasses.html; vs `@nt.utils.dataclasses.dataclasss`, as it was before), and avoid this kind of changes?

Reply:
So the problem here is not about `dataclasses`. It is that `dataclasses.asdict` method will trigger deepcopy of the objects, where TF Tensor does not allow being deep-copied.

Reply:
Ok, but perhaps then still you can keep it a `dataclasses.dataclass`, and provide it with a custom `Kernel._asdict()` method that would do wha you are doing here?

Reply:
Sounds good!